PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['crm.lead']
-    _inherit = ['mailing.mailing']
-    _inherit = ['utm.campaign']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mass_mailing_crm/models/crm_lead.py
+++ b/addons/mass_mailing_crm/models/crm_lead.py
@@ -5,5 +5,5 @@ from odoo import models
 
 
 class CrmLead(models.Model):
-    _inherit = ['crm.lead']
+    _inherit = 'crm.lead'
     _mailing_enabled = True

--- a/addons/mass_mailing_crm/models/mailing_mailing.py
+++ b/addons/mass_mailing_crm/models/mailing_mailing.py
@@ -6,7 +6,7 @@ from odoo import fields, models, _, tools
 
 
 class MailingMailing(models.Model):
-    _inherit = ['mailing.mailing']
+    _inherit = 'mailing.mailing'
 
     use_leads = fields.Boolean('Use Leads', compute='_compute_use_leads')
     crm_lead_count = fields.Integer('Leads/Opportunities Count', compute='_compute_crm_lead_count')

--- a/addons/mass_mailing_crm/models/utm.py
+++ b/addons/mass_mailing_crm/models/utm.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class UtmCampaign(models.Model):
-    _inherit = ['utm.campaign']
+    _inherit = 'utm.campaign'
 
     ab_testing_winner_selection = fields.Selection(selection_add=[('crm_lead_count', 'Leads')])
